debug.diff

```diff
@@ -0,0 +1,1191 @@
+1573,1597c1573,1598
+<       DEBUG assumptions={}, data={u32 = !ty_1, Not(!ty_1)}
+<       prove goal={u32 = ?ty_0, Not(?ty_0)}, assumptions={}, env=Env { variables: [?ty_0], bias: Completeness }, decls=decls(222, [trait Foo <ty> , trait Not <ty> ], [impl Foo(u32), impl <ty> Foo(^ty0_0) where {Not(^ty0_0)}], [], [], [], [], {Foo, Not}, {})
```

Comment:
in main, the `ty = var` is checked first (and probably instantiates the variable for the later check)